content/docs/command-reference/exp/run.md

```diff
@@ -0,0 +1,101 @@
+# exp run
+
+Run an experiment: reproduce a variation of a committed
+[pipeline](/doc/command-reference/dag) in a hidden project branch.
+
+> Similar to `dvc repro` but for
+> [experimentation](/doc/user-guide/experiment-management).
+
+## Synopsis
+
+```usage
+usage: dvc exp run [-h] [-q | -v] [-f]
+                   [<repro_options> ...]
+                   [--params [<filename>:]<params_list>]
+                   [-n <name>] [--queue] [--run-all] [-j <number>]
+                   [targets [targets ...]]
+
+positional arguments:
+  targets               Stages to reproduce. 'dvc.yaml' by default.
+```
+
+## Description
+
+Provides a way to execute and track `dvc experiments` in your
+<abbr>project</abbr> without polluting it with unnecessary commits, branches,
+directories, etc.
+
+> `dvc exp run` is equivalent to `dvc repro` for <abbr>experiments</abbr>. It
+> has the same behavior when it comes to stage execution (restores the
+> dependency graph, etc.). It differs from in that experiment dependencies and
+> outputs will always be <abbr>cached</abbr> (so they can be restored later).
+
+Each `dvc exp run` creates a variation based on the last project version (Git
+commit) and stores it internally with an automatic experiment ID (a short SHA-1
+hash). The results of the last experiment can be seen in the
+<abbr>workspace</abbr>. To display and compare your experiments, use
+`dvc exp show` or `dvc exp diff`.
+
+<details>
+
+### How does DVC track experiments?
+
+`dvc exp` uses actual commits under custom
+[Git references](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
+(found in `.git/refs/exps`). Each commit has the Git `HEAD` as parent. These are
+not pushed to the Git remote by default (see `dvc exp push`).
+
+> References have a unique signature similar to the
+> [entries in the run-cache](/doc/user-guide/project-structure/internal-files#run-cache).
+
+</details>
+
+Successful experiments can be made
+[persistent](/doc/user-guide/experiment-management#persistent-experiments) by
+committing them to the Git repo. You can use `dvc exp apply` to roll back the
+workspace to a previous experiment first. Unnecessary ones can be removed with
+`dvc exp gc`, or abandoned (and their data with `dvc gc`).
+
+## Parallel execution (multi-threading)
+
+⚠️ Multi-thread runs are experimental and may be unstable. ⚠️
+
+When using `-j` to run experiments in parallel, take caution to only specify a
+number of jobs which can be handled by your local machine or execution
```

Comment:
What's parallelized into jobs exactly, if stages are not scheduled? Does -j only affect independent stages?

Reply:
The entire pipeline is what gets parallelized.

The existing behavior equivalent for `-j 2` would be cloning your git repo twice (into two separate folders/workspaces), and then doing `dvc repro ...` in each one

Reply:
> cloning your git repo twice (into two separate folders/workspaces), and then doing dvc repro ... in each one

That sounds like duplicating the work. I'm still not getting this 🤔

Reply:
It would be duplicating the work if the user queued two identical experiments, but the main use case is to queue up multiple experiments with different parameter values (or different versions of code/dependencies/etc) and then run those different experiments simultaneously. If you were going to do this in a single DVC/git clone/workspace with `dvc repro`, you would have to run all of your experiments one at a time sequentially.

Reply:
Also, this should probably not be documented as `multithreading`. I would consider just using `parallel runs` everywhere in place of multithreading.

The original implementation was to do runs in separate threads, but the current implementation is via multiprocessing - so each run is in its own process, but this is also an internal implementation detail that is subject to change in the future.

Reply:
> -j/--jobs is only applicable for exp run --run-all

Got it. So could it maybe be an optional `<number>` argument to `--run-all` instead of a separate flag?

> this should probably not be documented as multithreading... the current implementation is via multiprocessing

Got it, thanks. I changed the language.

Reply:
> So could it maybe be an optional <number> argument to --run-all instead of a separate flag?

Maybe? Using `--jobs` seems more consistent with other DVC commands to me.

Reply:
It does seem that way.